standards/bulk.md

```diff
@@ -0,0 +1,134 @@
+- Bulk operations **MUST** accept a constrained number of operations in the request body that is indicated in the documentation of the endpoint. By default this value **MAY** be 100 operations, but should be adjusting according to the needs of the endpoint and the entity-size. Requests beyond the limit **MUST** return a `400` response code and standard [error format body](errors.md) similar to:
+```
+// RESPONSE
+HTTP/1.1 400
+Content-Type: application/problem+json
+{
+    "title": "Invalid Data",
+    "status": 400,
+    "detail": "Operations collection may only contain a maximum of '100' actions per request.",
+    "instance": "/articles",
+    "requestId": "b6d9a290-9f20-465b-bcd3-4a5166eeb3d7"
+}
+```
+
+- Bulk operations **MUST** include a request body schema following: 
+```json
+{
+    "operations": [
+        {
+            "action": "CREATE" | "UPDATE" | "CREATE_UPDATE" | "DELETE",     // indicate intent of operation
+            "ifMatch": "string" | null,                                     // if-match is an optional ETag that can be passed for optimistic concurrency
+            "entity": {
+                ... MATCHING ENTITY...                                      // must match entity schema resource from the collection
+            }                                                               // schema on entity is static and not dynamic
+        },
+    ]
+}
+```
+
+- Bulk operations **MUST** include a response body schema following:
+```json
+{
+    "status": "SUCCEEDED" | "FAILED" | "PARTIAL",                           // overall status of bulk operation, partial indicating 
+    "operations": [                                                         // there are some operations that failed and succeeded
+        {
+            "ordinal": 0,                                                   // position in original ordered operation collection
+            "action": "CREATE" | "UPDATE" | "CREATE_UPDATE" | "DELETE",     // repeat action type
+            "id": "string" | null,                                          // the associated id of the entity, if available
+            "ref": "sps-ref" | null,                                        // the associated sps-ref URN entity, if applicable
+            "status": "SUCCEEDED" | "FAILED",                               // status of individual operation
+            "detail": "string",                                             // indicates details on the operation result, such as a error message.
+        }
+    ]
+}
+```
+
+The following example demonstrates various operations in a bulk request and response:
+
+```
+// REQUEST
+PATCH /articles
+Content-Type: application/json
+{
+    "operations": [
+        {
+            "action": "CREATE_UPDATE"
+            "ifMatch": "33a64df551425fcc55e4d42a148795d9f25f89d4",
```

Comment:
Is it a need to handle optimistic concurrency in bulk? seems relevant... but an etag for each entity is necessary then. Should this field name match the "If-Match" header or just call it "etag"?

Reply:
And an extra wrinkle that it doesn't apply in every scenario, such as CREATE. (Maybe also not in DELETE either, unless standards changed there since I last looked?)

I like the idea of feature parity as if you called these all individually, though.

Reply:
I assumed it was an optional field not required, so don't provide on create. It would apply to DELETE if you wanted it too? Was there a specific callout on why not to use an if-match on delete if you needed it?

Reply:
I just wasn't sure what our standards said. I thought it was just a PUT/PATCH thing. DELETE could make sense, but I don't know if that's part of what we say you should be doing. 🤷‍♂️

Reply:
I like this.  It's a good catch to notice it and include it.
Do we only support if-match and etags?  Isn't there a version for date/time, as well?  do our standards ignore that?

Reply:
> DELETE could make sense, but I don't know if that's part of what we say you should be doing.
If someone wanted to ensure concurrency when they DELETED something I have no problem with the DELETE accepting the `if-match` header. I don't think that is outside the api standards. Use case specific.

> Do we only support if-match and etags? Isn't there a version for date/time, as well? do our standards ignore that?
We have only so far supported ETags... is there something specific you had in mind for the version of this with date/time? Is it worth investigating further? I can see that being something we easily add later if we wanted to more holistically support it. But have not used it to date that I'm aware of...